code/modules/mob/living/living.dm

```diff
@@ -1,7 +1,9 @@
 /mob/living/Initialize(mapload)
 	. = ..()
 	if(current_size != RESIZE_DEFAULT_SIZE)
-		update_transform(current_size)
+		var/cached_size = current_size
+		current_size = RESIZE_DEFAULT_SIZE
+		update_transform(cached_size)
```

Comment:
the issue is current size is used as a template var and as something that does something, right? would it not be better to split it up into two?

Reply:
I want to disagree. This isn't a template for a instanced datum but just a number that keeps track of the current size of the mob. I think it'd be faily ridicolous to have two vars with exact same purpose.

Reply:
one is the current size of the mob the other is the current size we want to apply at init, aren't those different things? that is what's going on here right?

Reply:
They're both "current size" tho.

Reply:
one is the current size it WILL be, the other is the current size it IS, right?
like if you have it the same you risk this same class of problem everywhere right? seems silly to me

Reply:
Fine, let's split it in two.